src/main/scala/com/bazaarvoice/sswf/model/StepResults.scala

```diff
@@ -0,0 +1,91 @@
+package com.bazaarvoice.sswf.model
+
+sealed abstract class StepResult(message: Option[String]) {
+  def isSuccessful: Boolean
+  def isInProgress: Boolean
+}
+
+case class Success(message: Option[String]) extends StepResult(message) {
+  def this() = this(None)
```

Comment:
is this just so Java folks can new it up? I'm guessing scala objects don't directly translate to static factories by default (i.e., if you import this guy in Java, you can't say "Success(stringOrNull)" and get a "new Success(stringOrNull)" back.